src/pages/checkout/Checkout.jsx

```diff
@@ -0,0 +1,168 @@
+import React from "react";
+import { useNavigate, Link } from "react-router-dom";
+import { useAuth, useCart, useOrder, useData } from "../../context";
+import "./checkout.css";
+import { CheckoutAddress } from "./CheckoutAddress";
+import { toast } from "react-toastify";
+import { ErrorIcon } from "../../assets";
+
+const Checkout = () => {
+  const navigate = useNavigate();
+  const {
+    token,
+    user: { firstName, lastName, email },
+  } = useAuth();
+  const { priceDetails, orderDispatch, setOrder, orderAddress } = useOrder();
+  const { cartTotal, offerDiscount, couponDiscount, totalAmt } = priceDetails;
+  const { cartList, clearCart, cartDispatch } = useCart();
+  const { address } = useData();
+
+  const loadScript = async (url) => {
+    return new Promise((resolve) => {
+      const script = document.createElement("script");
+      script.src = url;
+      script.onload = () => {
+        resolve(true);
+      };
+      script.onerror = () => {
+        resolve(false);
+      };
+      document.body.appendChild(script);
```

Comment:
Not sure whether this is the right approach like using vanilla JS methods in React. Are you using this razorpay library for payment?

Reply:
Yes.  I am using razorpay for payment. And I got it to work in this way only .